server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregator.java

```diff
@@ -64,11 +64,18 @@
     private int targetBuckets;
     private MergingBucketsDeferringCollector deferringCollector;
 
-    AutoDateHistogramAggregator(String name, AggregatorFactories factories, int numBuckets, RoundingInfo[] roundingInfos,
-            @Nullable ValuesSource.Numeric valuesSource, DocValueFormat formatter, SearchContext aggregationContext, Aggregator parent,
-            List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) throws IOException {
-
-        super(name, factories, aggregationContext, parent, pipelineAggregators, metaData);
+    AutoDateHistogramAggregator(AggregatorFactories factories,
+                                int numBuckets,
+                                String minimumIntervalExpression,
+                                RoundingInfo[] roundingInfos,
+                                @Nullable ValuesSource.Numeric valuesSource,
+                                DocValueFormat formatter,
+                                SearchContext aggregationContext,
+                                Aggregator parent,
+                                List<PipelineAggregator> pipelineAggregators,
+                                Map<String, Object> metaData) throws IOException {
+
+        super(minimumIntervalExpression, factories, aggregationContext, parent, pipelineAggregators, metaData);
```

Comment:
The aggregator name seems to have been replaced by the `minimumIntervalExpression`, the `name` arg seems to have been removed entirely.

Also, it's not clear to me why the `minimumIntervalExpression` needs to be passed to the `AutoDateHistogramAggregator`?